tests/cecil-tests/ApiTest.cs

```diff
@@ -49,5 +49,116 @@ static void AddFailure (ref List<string>? failures, string failure)
 			failures.Add (failure);
 			Console.WriteLine (failure);
 		}
+
+		// Capitalization rules:
+		// * All types, methods, properties, fields, events must start with an upper-cased letter.
+		// * All parameters must start with a lower-cased letter
+		[Test]
+		public void MustStartWithCapitalLetter ()
+		{
+			Configuration.IgnoreIfAnyIgnoredPlatforms ();
+
+			var failures = new Dictionary<string, (string Message, string Location)> ();
+
+			foreach (var info in Helper.NetPlatformImplementationAssemblyDefinitions) {
+				var assembly = info.Assembly;
+				foreach (var member in assembly.EnumeratePublicMembers ()) {
+					// Presumably obsolete members have been obsoleted for a reason, so assume there's a correctly capitalized version.
+					if (((ICustomAttributeProvider) member).IsObsolete ())
+						continue;
+
+					if (member is MethodDefinition method) {
+						// skip constructors
+						if (method.IsConstructor)
+							continue;
+						// skip property accessors
+						if (method.IsPropertyAccessor ())
+							continue;
+						// skip event methods
+						if (method.IsEventMethod ())
+							continue;
+						// skip operators
+						if (method.IsOperator ())
+							continue;
+					} else if (member is FieldDefinition field) {
+						if (field.Name == "value__" && !field.IsStatic && field.DeclaringType.IsEnum)
+							continue;
+					}
+
+					if (!char.IsUpper (member.Name [0]) && !IsAcceptableCapitalization (member)) {
+						var msg = $"The {member.GetType ().Name.Replace ("Definition", "").ToLower ()} '{member.FullName}' has incorrect capitalization: first letter is not upper case.";
+						failures [msg] = new (msg, member.RenderLocation ());
+					}
+				}
+
+				foreach (var method in assembly.EnumerateMethods (Helper.IsPubliclyVisible)) {
```

Comment:
It looks like `EnumeratePublicMembers` already hands you methods. Maybe you want to move the foreach on the parameters up above and add an `else if (member is MethodDefinition method) { }` around it.


Also, will this work in inner classes?

Reply:
@stephen-hawley should be fixed now